app/src/main/java/app/revanced/integrations/patches/components/AdsFilter.java

```diff
@@ -0,0 +1,265 @@
+package app.revanced.integrations.patches.components;
+
+
+import android.view.View;
+import app.revanced.integrations.settings.SettingsEnum;
+import app.revanced.integrations.utils.LogHelper;
+import app.revanced.integrations.utils.ReVancedUtils;
+
+
+public final class AdsFilter extends Filter {
+    private final String[] exceptions;
+
+    private final CustomFilterGroup custom;
+
+    public AdsFilter() {
+        exceptions = new String[]{
+                "home_video_with_context",
+                "related_video_with_context",
+                "comment_thread", // skip filtering anything in the comments
+                "|comment.", // skip filtering anything in the comments replies
```

Comment:
Please add `playlist_add_to_option_wrapper` to `exceptions`, sometimes it shows and sometimes video is added to random playlist without showing playlist flyout panel.

Reply:
> sometimes it shows

What shows?

Reply:
> > sometimes it shows
> 
> What shows?

Playlist flyout panel.

Reply:
How is that caused by the filter? This behavior is stock behavior, if you click on the playlist button, it will add to the last selected playlist, if you hold click it'll open the playlist menu.

Reply:
When you click add to playlist in feed videos or related videos, it should ask for playlist.

Reply:
It happens like 1 in 100 times, so it's difficult to reproduce.

Reply:
In that case, how can you be sure it is caused by a filter?

Reply:
In that case, how can you be sure it is caused by a filter?

Reply:
In that case, how can you be sure it is caused by a filter?

Reply:
Sounds good, it's not part of this PR anyways.